app/dao/jobs_dao.py

```diff
@@ -108,6 +109,8 @@ def dao_get_future_scheduled_job_by_id_and_service_id(job_id, service_id):
 
 
 def dao_create_job(job):
+    if not job.id:
+        job.id = uuid.uuid4()
```

Comment:
mainly because this felt like a reasonable time to provide it, so that anywhere else we use `dao_create_job` is handled as well